geo.go

```diff
@@ -0,0 +1,138 @@
+package tuna
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const IP2CUrl = "https://ip2c.org/"
+
+type Location struct {
+	IP          string `json:"IP"`
+	CountryCode string `json:"CountryCode"`
+	Country     string `json:"Country"`
+	City        string `json:"City"`
+}
+
+var emptyLocation = Location{}
+
+func (l *Location) Empty() bool {
+	if l == nil {
+		return true
+	}
+	return *l == emptyLocation
+}
+
+type IPFilter struct {
+	Allow    []Location `json:"Allow"`
+	Disallow []Location `json:"Disallow"`
+}
+
+func (f *IPFilter) Empty() bool {
+	if f == nil {
+		return true
+	}
+
+	for _, loc := range f.Allow {
+		if !loc.Empty() {
+			return false
+		}
+	}
+	for _, loc := range f.Disallow {
+		if !loc.Empty() {
+			return false
+		}
+	}
+	return true
+}
+
+func (f *IPFilter) IPCheck(ip string) bool {
+	if f.Empty() {
+		return true
+	}
+	for _, loc := range f.Disallow {
+		if len(loc.IP) == 0 {
+			continue
+		}
+		if loc.IP == ip {
+			return false
+		}
+	}
+	for _, loc := range f.Allow {
+		if len(loc.IP) == 0 {
+			continue
+		}
+		if loc.IP == ip {
+			return true
+		}
+		return false
+	}
+
+	return true
```

Comment:
If we return true at the end of the function, then f.Allow will not have any effect (only Disallow will be used).

Instead, I think we should do IPCheck and CountryCheck together for each Location.